mets_mods2tei/scripts/update.py

```diff
@@ -29,7 +29,7 @@ class WorkspaceCtx():
     def __init__(self, directory, mets_url, automatic_backup):
         self.log = getLogger('mets_mods2tei.update')
         self.resolver = Resolver()
-        self.directory, self.mets_url, self.mets_basename = self.resolver.resolve_mets_arguments(directory, mets_url, None)
+        self.directory, self.mets_url, self.mets_basename, self.mets_server_url = self.resolver.resolve_mets_arguments(directory, mets_url, None, None)
```

Comment:
Unless you want to make use of it, it's probably better not to declare the variable at all.

```suggestion
        self.directory, self.mets_url, self.mets_basename, _= self.resolver.resolve_mets_arguments(directory, mets_url, None, None)
```

If you want to be fully compatible with the behavior of the `ocrd workspace` group, consider adding the additional parameters from https://github.com/OCR-D/core/blob/master/ocrd/ocrd/cli/workspace.py#L45-L51 and `from ocrd.cli.workspace import WorkspaceCtx` and subclassing that.

Just an idea, no need to change anything.

Reply:
Did that cause i was getting the error "ValueError: too many values to unpack"

Reply:
Sure, you need something to balance left- and right-hand side of the assignment, but you can use `_` to skip assignment to `self.mets_server_url` is what I meant.

Reply:
I agree, we should anonymize the new value.

Reply:
> If you want to be fully compatible with the behavior of the `ocrd workspace` group, consider adding the additional parameters from https://github.com/OCR-D/core/blob/master/ocrd/ocrd/cli/workspace.py#L45-L51 and `from ocrd.cli.workspace import WorkspaceCtx` and subclassing that.

@kba, indeed, directly reusing `ocrd.cli.workspace.WorkspaceCtx` (or even better `ocrd.cli.workspace.pass_workspace`) seems better. So can we consider this stable API (despite being part of the CLI subpackage)?

Reply:
IINM `WorkspaceCtx`'s constructor should use keyword arguments instead of positional ones, because otherwise you are forced to provide exactly the same CLI arguments to reuse that class. (But I don't want to have `mets_basename` nor `mets_server_url` in mm-update.)

But can I at least assume that `ocrd.Resolver.resolve_mets_arguments` will be kept compatible henceforth? (IMO the new `mets_server_url` there should have been added as a kwarg as well.)

Reply:
> > IINM WorkspaceCtx's constructor should use keyword arguments instead of positional ones, because otherwise you are forced to provide exactly the same CLI arguments to reuse that class. (But I don't want to have mets_basename nor mets_server_url in mm-update.)
> 
> I don't understand the distinction in this case, since all `__init__` args are required. But we do use them as kwargs. Or do you mean you want defaults for `mets_basename` and `mets_server_url`, so you can use just the ones you need?

Yes, that's what I said. But reusing that class is a very special case anyway, so never mind!



> Those changes, to `WorkspaceCtx` and `resolve_mets_arguments` would be the last ones in the foreseeable future, otherwise the API should be stable.

Understood – thx!